src/v/cluster/id_allocator_stm.h

```diff
@@ -34,10 +34,7 @@ class id_allocator_stm final : public raft::persisted_stm<> {
 public:
     static constexpr std::string_view name = "id_allocator_stm";
 
-    struct stm_allocation_result {
-        int64_t id;
-        raft::errc raft_status{raft::errc::success};
-    };
+    using stm_allocation_result = result<int64_t, raft::errc>;
```

Comment:
nit: I think this change can be made a separate commit

(to keep our Git history clear and easy to understand, we try to break PRs into smaller compilable commits, also helps the reviewers)

Reply:
If I do this I'll have to refactor the old test to use the new return type. I will do if it is not much work.